WIP: Add typing hint for src codes

Yanqiao4396 opened the pull request:
<!-- TODO: Replace the title below -->
# Your Title
Add typing hint
## Description

<!-- TODO: Write a description of the proposed changes -->
<!-- Remember to check the reminders! -->
Add the typing hint for all the I/O of functions in source codes
## Linked Issues

<!-- Fill in which issue number this PR closes, if there is none, just remove this section! -->
closes: #93\

## Type of Change

<!-- TODO: Fill in the brackets with an `x` next to all types that apply to the proposed changes -->
- [x] Feature
- [ ] Bug fix
- [ ] Documentation

## Contributors

<!-- TODO: Add your GitHub username below and the GitHub usernames of all other contributors to the proposed changes -->
- @Yanqiao4396 

## Reminder

 - All GitHub Actions should be in a passing state before any pull request is merged.
 - All PRs must be reviewed by at least one team member and one member of the Integration team!
 - Any issues this PR closes are tagged in the description!

Yanqiao4396 committed:
Add typing hint for src codes

gkapfham commented:
Hello @Yanqiao4396 if you want to continue to work on this PR, that would be great!

I encourage you to integrate `mypy` type checking into the GitHub Actions configuration, if it is not there already, so as to ensure that there is an automated tool to confirm that all of your type checks are correct.

Let's continue the conversation here! I am glad to help out with and/or review this PR as needed.

Thanks!

Yanqiao4396 committed:
Add mypy checker and edit lint cmd

Add mypy library. Put mypy into lint
check. Choose to ignore importing warning because the project works in
virtual environment

Yanqiao4396 committed:
Add typing hint to variables

Yanqiao4396 committed:
Fix some typing mistakes

Yanqiao4396 committed:
Add typing for main.py

Yanqiao4396 committed:
Add poetry.lock file

Yanqiao4396 committed:
Add type ignore

Mypy expects to find all the dependencies. However, we are building
on virtual environment, so ignore Mypy
"Library stubs not installed" message

Yanqiao4396 committed:
Black check

Yanqiao4396 commented:
The Mypy is added successfully. The related changes pass!!

gkapfham commented:
I have integrated three different type checkers into the current version of the tool and also confirmed that they are all passing correctly. Everything is now run through `uv`. I am going to close this PR as there are conflicts and the overall features are now provided.
